Content.Server/Fluids/EntitySystems/PuddleSystem.Evaporation.cs

```diff
@@ -33,30 +35,41 @@ private void UpdateEvaporation(EntityUid uid, Solution solution)
     private void TickEvaporation()
     {
         var query = EntityQueryEnumerator<EvaporationComponent, PuddleComponent>();
-        var xformQuery = GetEntityQuery<TransformComponent>();
         var curTime = _timing.CurTime;
         while (query.MoveNext(out var uid, out var evaporation, out var puddle))
         {
             if (evaporation.NextTick > curTime)
                 continue;
 
-            evaporation.NextTick += EvaporationCooldown;
+            var tileMix = _atmosphereSystem.GetContainingMixture(uid, true);
+            var hasEnoughVapor = tileMix?.GetMoles(Gas.WaterVapor) >= 1.0f;
 
-            if (!_solutionContainerSystem.ResolveSolution(uid, puddle.SolutionName, ref puddle.Solution, out var puddleSolution))
-                continue;
-
-            var reagentTick = evaporation.EvaporationAmount * EvaporationCooldown.TotalSeconds;
-            puddleSolution.SplitSolutionWithOnly(reagentTick, EvaporationReagents);
+            if (hasEnoughVapor)
+                EnsureComp<PreventEvaporationComponent>(uid).Active = true;
+            else
+                RemComp<PreventEvaporationComponent>(uid);
 
-            // Despawn if we're done
-            if (puddleSolution.Volume == FixedPoint2.Zero)
+            if (!hasEnoughVapor)
             {
-                // Spawn a *sparkle*
-                Spawn("PuddleSparkle", xformQuery.GetComponent(uid).Coordinates);
-                QueueDel(uid);
-            }
+                evaporation.NextTick = _timing.CurTime + EvaporationCooldown;
+                Dirty(uid, evaporation);
 
-            _solutionContainerSystem.UpdateChemicals(puddle.Solution.Value);
+                if (_solutionContainerSystem.ResolveSolution(uid, puddle.SolutionName, 
+                    ref puddle.Solution, out var puddleSolution))
```

Comment:
You can invert this check.
```suggestion
                if (!_solutionContainerSystem.ResolveSolution(uid,
                        puddle.SolutionName,
                        ref puddle.Solution,
                        out var puddleSolution))
                    continue;
```

Reply:
Fixed in one of the commits